packit_service/worker/whitelist.py

```diff
@@ -39,33 +40,41 @@
 
 
 class Whitelist:
-    def __init__(self):
+    def __init__(self, fas_user: str = None, fas_password: str = None):
         self.db = PersistentDict(hash_name="whitelist")
+        self._fas: AccountSystem = AccountSystem(
+            username=fas_user, password=fas_password
+        )
 
-    @staticmethod
-    def _is_packager(account_login: str) -> bool:
+    def _is_packager(self, account_login: str) -> bool:
         """
-        If GitHub username is same as FAS username this method checks if user is packager.
-        User is considered to be packager when he/she has the badge:
-         `If you build it... (Koji Success I)`
+        Check if the user is a packager, by checking if their GitHub
+        username is in the 'packager' group in FAS. Works only the user's
+        username is the same in GitHub and FAS.
+
         :param account_login: str, Github username
         :return: bool
         """
 
-        url = f"https://badges.fedoraproject.org/user/{account_login}/json"
-        response = requests.get(url)
-        if not response.status_code == 200:
-            if response.status_code == 404:
-                logger.info(f"No FAS {account_login!r} in badges.fedoraproject.org.")
-            else:
-                logger.error(f"Failed to get {account_login!r} from badges.fp.org.")
+        try:
+            person = self._fas.person_by_username(account_login)
+        except AuthError as e:
+            logger.error("FAS authentication failed: %s", e)
+            return False
+        except FedoraServiceError as e:
+            logger.error("FAS query failed: %s", e)
+            return False
+
+        if not person:
+            logger.info("Not a FAS username %r.", account_login)
             return False
 
-        for item in response.json().get("assertions", []):
-            if "Successfully completed a koji build." in item.get("description"):
-                logger.info(f"User {account_login!r} is a packager in Fedora!")
+        for membership in person.get("memberships", []):
+            if membership.get("name") == "packager":
+                logger.info("User %r is a packager in Fedora!", account_login)
                 return True
-        logger.info(f"Cannot verify whether {account_login!r} is a packager in Fedora.")
+
+        logger.info("Cannot verify whether %r is a packager in Fedora.", account_login)
```

Comment:
There's nothing wrong with you string formatting, it's just that we use the f-strings everywhere (even in logging), so I'd rather see them used also in your PR.

Reply:
Yes, I'm aware that this goes against the conventions of the code-base. Still I think this is better, b/c then string interpolation is left up to `logging`, and it's done only when the log-level requires it (less work). Plus aggregation becomes easier. 

I fail to find the relevant docs to point at right now, but [this post and comment](https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html#comment-2063573645) maybe explains it better.

Anyways, if you prefer consistency, just let me know and I'll change them to f-strings.

Reply:
Yes, we know about that (that was my `(even in logging)` comment supposed to tell you) and we still prefer consistency. I'm just not sure what you mean with 'aggregation'.

Reply:
Updated to f-strings.

By aggregation I meant, that for example Sentry would create a single event for `Cannot verify whether %r is a packager in Fedora.` repeating n times. But this might be achievable in other ways, too (~[Server-side fingerprinting](https://docs.sentry.io/data-management/event-grouping/server-side-fingerprinting/)).